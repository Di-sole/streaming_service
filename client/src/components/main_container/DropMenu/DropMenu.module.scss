.container {
    position: absolute;
    top: 67px;
    left: 0;
    z-index: 1;
    background-color: #1f1b2e;
    transition: all 0.4s;
    border-radius: 0 0 15px 15px;
    border-top: 1px solid #343434;
    padding: 30px 10px 10px 10px;
    display: flex;
    flex-wrap: wrap;
    height: 300px;
    width: 100%;

    &__item {
        display: flex;
        flex-direction: column;
        height: 100%;
        &:first-child {
            width: 450px;
        }
        &:not(:first-child) {
            width: 200px;
        }
    }

    &__title {
        color: #fff;
        margin: 0 0 15px 0;
        font-weight: 700;
    }

    &__content {
        display: flex;
        flex-direction: column;
        flex-wrap: wrap;
        height: 90%;
    }

    &__link {
        margin: 0 0 5px 0;
        font-weight: 500;
        width: max-content;
        padding-right: 20px;
    }
}
