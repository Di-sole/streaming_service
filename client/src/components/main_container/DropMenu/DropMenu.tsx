import styles from './DropMenu.module.scss';
import TextLinkUI from '../../UI/links/TextLink/TextLinkUI';
import { firstCapitalLetter } from '../../../utils/functions';

interface IContent {
    title: string;
    links: {
        text: string;
        href: string;
    }[];
}

interface DropMenuProps {
    className: string;
    content: IContent[];
}

const DropMenu = ({ className, content }: DropMenuProps) => {
    return (
        <div className={[styles.container, className].join(' ')} data-testid={'dropMenu'}>
            {content.map((item) => (
                <div key={item.title} className={styles.container__item}>
                    <h2 className={styles.container__title}>{item.title}</h2>
                    <div className={styles.container__content}>
                        {item.links.map((link, index) => (
                            <TextLinkUI key={index} className={styles.container__link} href={link.href} option="dim">
                                {firstCapitalLetter(link.text)}
                            </TextLinkUI>
                        ))}
                    </div>
                </div>
            ))}
        </div>
    );
};

export default DropMenu;
