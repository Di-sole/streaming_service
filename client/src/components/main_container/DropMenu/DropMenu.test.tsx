import { render, screen } from '@testing-library/react';
import DropMenu from './DropMenu';

describe('DROPMENU SNAPSHOTS TESTS', () => {
    test('DropMenu should not be changed', () => {
        render(
            <DropMenu
                className="container__dropMenu"
                content={[
                    {
                        title: 'title',
                        links: [
                            {
                                text: 'Ссылка 1',
                                href: '/link1',
                            },
                            {
                                text: 'Ссылка 2',
                                href: '/link2',
                            },
                            {
                                text: 'Ссылка 3',
                                href: '/link3',
                            },
                        ],
                    },
                ]}
            />,
        );

        const div = screen.getByTestId('dropMenu');
        expect(div).toMatchSnapshot();
    });
});
