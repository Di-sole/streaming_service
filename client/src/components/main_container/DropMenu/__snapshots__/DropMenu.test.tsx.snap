// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DROPMENU SNAPSHOTS TESTS DropMenu should not be changed 1`] = `
<div
  class="container container__dropMenu"
  data-testid="dropMenu"
>
  <div
    class="container__item"
  >
    <h2
      class="container__title"
    >
      title
    </h2>
    <div
      class="container__content"
    >
      <a
        class="container container_dim container__link"
        href="/link1"
      >
        Ссылка 1
      </a>
      <a
        class="container container_dim container__link"
        href="/link2"
      >
        Ссылка 2
      </a>
      <a
        class="container container_dim container__link"
        href="/link3"
      >
        Ссылка 3
      </a>
    </div>
  </div>
</div>
`;
